include/dts-functions.sh

```diff
@@ -1815,5 +1727,35 @@ check_if_intel() {
   if [ $cpu_vendor == "GenuineIntel" ]; then
     return 0
   fi
+}
+
+parse_for_arg_return_next(){
```

Comment:
Wouldn't it be better to put in hal dir or maybe even `common-mock-func.sh` as it's only used there?

Reply:
I wanted to separate it. HAL directory is only meant for functions that do some hardware-related stuff (inluding mocking of these hardware-related stuff).

On the other hand, `dts-functions.sh` is a file with functions that do common work.

I agree, that currently, `dts-functions.sh` contain a bit of hardware-related funcs., but it will be changed when we will migrate to another language.

Reply:
> but it will be changed when we will migrate to another language

Hmm, later (i.e. never).

> On the other hand, dts-functions.sh is a file with functions that do common work

Well, it doesn't do common work as it's only used in `common-mock-func.sh`.
So it's helper function for tests/mocking, it probably should be there then. You can create `helper-functions.sh` there if you want. I'm pretty sure it's better than adding more and more functions into one file without end.

Reply:
Addressed [here](https://github.com/Dasharo/dts-scripts/compare/8e94cbc7f99ffe845b06f6f742c9b9863dec795e..992062280bbd17d42f45e14d98c3e682971d4ee4)